components/post_markdown/index.test.ts

```diff
@@ -4,25 +4,29 @@ import assert from 'assert';
 
 import {TestHelper} from 'utils/test_helper';
 
+import {GlobalState} from 'types/store';
+
 import {makeGetMentionKeysForPost} from './index';
 
 describe('makeGetMentionKeysForPost', () => {
     const channel = TestHelper.getChannelMock({});
     const team = TestHelper.getTeamMock({group_constrained: false});
-    const user = TestHelper.getUserMock({
+    let user = TestHelper.getUserMock({
         username: 'a123',
+    });
+    user = {...user,
         notify_props: {
+            ...user.notify_props,
             channel: 'true',
-        },
-    });
+        }};
     const group = TestHelper.getGroupMock({
-        id: 123,
+        id: '123',
         name: 'developers',
         allow_reference: true,
         delete_at: 0,
     });
 
-    const baseState = {
+    const baseState: GlobalState = {
```

Comment:
this initializing with the type is not needed, since we are type casting the variable anyways

Reply:
Is it doing any harm

Reply:
Just curious to know

Reply:
its just unnecessary to doubly do that.